notebook/tests/selenium/utils.py

```diff
@@ -0,0 +1,163 @@
+import os
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+
+from contextlib import contextmanager
+
+pjoin = os.path.join
+
+
+def wait_for_selector(browser, selector, timeout=10, visible=False, single=False):
+    wait = WebDriverWait(browser, timeout)
+    if single:
+        if visible:
+            conditional = EC.visibility_of_element_located
+        else:
+            conditional = EC.presence_of_element_located
+    else:
+        if visible:
+            conditional = EC.visibility_of_all_elements_located
+        else:
+            conditional = EC.presence_of_all_elements_located
+    return wait.until(conditional((By.CSS_SELECTOR, selector)))
+
+
+class CellTypeError(ValueError):
+    
+    def __init__(self, message=""):
+        self.message = message
+
+class Notebook:
+    
+    def __init__(self, browser):
+        self.browser = browser
+        self.remove_safety_check()
+
+    @property
+    def body(self):
+        return self.browser.find_element_by_tag_name("body")
+
+    @property
+    def cells(self):
+        """Gets all cells once they are visible.
+        
+        """
+        wait_for_selector(self.browser, ".cell")
```

Comment:
I don't like having a wait inside a property - the rule of thumb for properties is that anything they need to do should be near-instant. If it needs to do something that's not, let's make it a regular method - the call gives people reading the code a hint that there's something going on underneath.

Reply:
I was wrong in my belief that it was needed here. One of the reasons I really didn't like it was that it couldn't handle notebooks with no cells; so this always seemed wrong. So it's gone!